hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactNullablePrimitiveInteroperabilityTest.java

```diff
@@ -229,9 +289,45 @@ public void testWriteNullReadPrimitiveThrowsException() {
         assertReadNullAsPrimitiveThrowsException(serializedRecord);
     }
 
+    @Test
+    public void testWriteNullReadPrimitiveThrowsExceptionCustomSerializer() {
+        FixedFieldsDTO fixedFieldsDTO = createFixedFieldsDTO();
+        CompactSerializationConfig compactSerializationConfig = new CompactSerializationConfig();
+        compactSerializationConfig.addSerializer(new FixedFieldsDTOSerializerWritingNull());
+        SerializationService serializationService = new DefaultSerializationServiceBuilder()
+                .setSchemaService(schemaService)
+                .setConfig(new SerializationConfig().setCompactSerializationConfig(compactSerializationConfig))
+                .build();
+
+        Data data = serializationService.toData(fixedFieldsDTO);
+
+        assertThatThrownBy(() -> serializationService.toObject(data))
```

Comment:
I didn't like this test because the exception will be thrown the first time you try to read a primitive field with a null value. If that is okay then there is no need to write a special serializer for it. We can use the `ASerializer` above (I believe we use it for a similar test)

Reply:
Ok, reasonable. I did not like doing that either, but looks like there is no way to test all fixed size fields.

Reply:
Now, how is this test different than `testWriteNullReadPrimitiveThrowsExceptionWithCorrectMethodPrefixCompactReader`?

Reply:
yeah I will delete `testWriteNullReadPrimitiveThrowsExceptionCustomSerializer` then.